cmd/telemeter-server/main.go

```diff
@@ -275,212 +275,218 @@ func (o *Options) Run() error {
-	)
-	signer := jwt.NewSigner(issuer, privateKey)
+				key, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+				if err != nil {
+					return fmt.Errorf("key generation failed: %v", err)
+				}
 
-	external := http.NewServeMux()
-	internal := http.NewServeMux()
+				privateKey, publicKey = key, key.Public()
+			}
 
-	// configure the authenticator and incoming data validator
-	var clusterAuth authorize.ClusterAuthorizer = authorize.ClusterAuthorizerFunc(stub.Authorize)
-	if authorizeURL != nil {
-		clusterAuth = tollbooth.NewAuthorizer(o.Logger, &authorizeClient, authorizeURL)
-	}
+			// Configure the whitelist.
+			if len(o.WhitelistFile) > 0 {
+				data, err := ioutil.ReadFile(o.WhitelistFile)
+				if err != nil {
+					return fmt.Errorf("unable to read --whitelist-file: %v", err)
+				}
+				o.Whitelist = append(o.Whitelist, strings.Split(string(data), "\n")...)
+			}
+			for i := 0; i < len(o.Whitelist); {
+				s := strings.TrimSpace(o.Whitelist[i])
+				if len(s) == 0 {
+					o.Whitelist = append(o.Whitelist[:i], o.Whitelist[i+1:]...)
+					continue
+				}
+				o.Whitelist[i] = s
+				i++
+			}
+			whitelister, err := metricfamily.NewWhitelist(o.Whitelist)
+			if err != nil {
+				return err
+			}
 
-	auth := jwt.NewAuthorizeClusterHandler(o.Logger, o.PartitionKey, o.TokenExpireSeconds, signer, o.RequiredLabels, clusterAuth)
-	validator := validate.New(o.PartitionKey, o.LimitBytes, 24*time.Hour, time.Now)
+			const issuer = "telemeter.selfsigned"
+			const audience = "telemeter-client"
 
-	var store store.Store
-	u, err := url.Parse(o.ForwardURL)
-	if err != nil {
-		return fmt.Errorf("--forward-url must be a valid URL: %v", err)
-	}
-	store = forward.New(o.Logger, u, nil)
+			jwtAuthorizer := jwt.NewClientAuthorizer(
+				issuer,
+				[]crypto.PublicKey{publicKey},
+				jwt.NewValidator(o.Logger, []string{audience}),
+			)
+			signer := jwt.NewSigner(issuer, privateKey)
 
-	// Create a rate-limited store with a memory-store as its backend.
-	store = ratelimited.New(o.Ratelimit, store)
+			// configure the authenticator and incoming data validator
+			var clusterAuth authorize.ClusterAuthorizer = authorize.ClusterAuthorizerFunc(stub.Authorize)
+			if authorizeURL != nil {
+				clusterAuth = tollbooth.NewAuthorizer(o.Logger, &authorizeClient, authorizeURL)
+			}
 
-	transforms := metricfamily.MultiTransformer{}
-	transforms.With(whitelister)
-	if len(o.Labels) > 0 {
-		transforms.With(metricfamily.NewLabel(o.Labels, nil))
-	}
-	transforms.With(metricfamily.NewElide(o.ElideLabels...))
+			auth := jwt.NewAuthorizeClusterHandler(o.Logger, o.PartitionKey, o.TokenExpireSeconds, signer, o.RequiredLabels, clusterAuth)
+			validator := validate.New(o.PartitionKey, o.LimitBytes, 24*time.Hour, time.Now)
 
-	server := httpserver.New(o.Logger, store, validator, transforms)
-	receiver := receive.NewHandler(o.Logger, o.ForwardURL, prometheus.DefaultRegisterer)
+			var store store.Store
+			u, err := url.Parse(o.ForwardURL)
```

Comment:
this is not really specific to the v1 routes since it's used by v2 storage too, right? maybe we can move this flag parsing to ~ line 277 with some of the other parsing